### PR TITLE
Assimilation changes

### PR DESCRIPTION
**Pull request by Kierany9**
[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. This includes, new features, sprites, sounds, balance changes, admin tools, map edits, removals, big refactors, config changes, hosting changes and important fixes. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs)

:cl: Kierany9
code: Several minor updates to Assimilation.
balance: Hosts no longer lose abilities if they fall below the required hive size.
balance: Mind Control victims can no longer see nor hear for the duration of the control.
tweak: Distortion Field now affects the victim and those surrounding them over time instead of instantly.
fix: Abilities that inflict stamina damage now affect the head instead of spreading damage over the entire body.
tweak: Mind Control now informs you of how long you have control for upon activation.
tweak: Changed various do_mob for do_after. All hivemind abilities except for Mass Assimilation should now work in certain situations where they didn't before.
balance: Abusing the sleep verb while mind controlling is no longer a viable tactic.
balance: Assimilating/removing people into/from the hive now has screen-wide range, with time taken to assimilate increasing exponentially with distance. The time to assimilate from 4 tiles away is approximately the same. Assimilation's cooldown has also been increased by five seconds to compensate.
/:cl:

Minor additions, changes and fixes for Assimilation that should handle some of the more nitpicky and learning-curve issues I've seen with it. Bigger changes coming ~~soon~~later:tm:.

~~(dnm, going to make a couple of things more sensible)~~

### CODE REVIEW

**Comment by Vidiveni** [comment]
Mind control, Medullary Failure, Mass Assimilation, and maybe some others, should give a ghost alerts imo. We're probably missing so many fun moments since ghosts won't notice any of the things happening.

**Comment by anconfuzedrock** [comment]
Mind control is going to be extremely awkward until hiveminds can cancel it at will or at least know when it runs out.

**Comment by zxaber** [comment]
>balance: Assimilating people into the hive now has screen-wide range, with time taken to assimilate increasing exponentially with distance. The time to assimilate from 4 tiles away is approximately the same. 

How does assimilating work, anyway? Does it just require both parties to stand still? Getting thermals as a hivemind is gonna make assimilating super easy.

**Comment by Kierany9** [comment]
> How does assimilating work, anyway? Does it just require both parties to stand still? Getting thermals as a hivemind is gonna make assimilating super easy.

Requires the hivemind to stand still, and for the other party to stay in range. Assimilation isn't meant to be a challenge, it's more of a timed gating mechanism to keep the power balance in check.

File: code/modules/spells/spell_types/hivemind.dm [hunk 1]
```diff
@@ -261,6 +261,8 @@
 			backseat.ghostize(0)
 		else
 			backseat.mind.transfer_to(vessel,1)
+			vessel.SetSleeping(0)
```
[the remark this answers is not in the document]
Comment (Kierany9) [reply code/modules/spells/spell_types/hivemind.dm]: Probably is, going to change this to something more sensible when I can.

**Comment by Kierany9** [comment]
> Why was the "you're not allowed to know what you did while mind-controlled" thing added? Do you still know that you were mind-controlled?

Mostly to be the same as other forms of antagonistic behaviour, like not being able to use ghost knowledge or not knowing anything that happened since conversion once deconverted from a cult, and to curb powergaming off of that knowledge. In retrospect I'll be removing it and add a couple of mechanical restrictions, mostly because it's unenforcable in most scenarios and the people who are going to ignore it anyway are the ones who are supposed to be curbed by it.

**Comment by anconfuzedrock** [comment]
Yeah while we’re at it terry should say “100% LAG FREE!” In font 500, big text is why internal affairs is such a beloved gamemode after all.

**Comment by Kierany9** [comment]
Changed the things I said I'd change. Instead of waking up after being mind control, sleep is mirrored to the host, and instead of the big "do not do the thing" text after being mind controlled, you won't be able to see or hear what's going on around you for the duration of the control.

**Comment by zxaber** [comment]
I'm not too familiar with the gamemode; there's no tell to the victim that they're being assimilated, right? I dunno, I think it's gonna feel terrible that someone can force-convert you, at range, with no requirement that you hold still, and without you even knowing. Just about every other mechanic in the game that can result in lasting effects has a tell or warning somewhere (virus symptoms have warning text, sleepy pen has the "feel a prick" text, flashes are visible when carried, etc).

It's one thing for there to be no warning when you have to be adjacent. Removing that restriction without adding any seems a bit too much.

**Comment by Kierany9** [comment]
> It's one thing for there to be no warning when you have to be adjacent. Removing that restriction without adding any seems a bit too much.

Being adjacent was never a restriction, the range has always been four tiles.

>I think it's gonna feel terrible that someone can force-convert you [...]

This is not a conversion mode.

**Comment by Ratvar** [comment]
Think about it more like a Tag game @zxaber

**Pull request merged**